src/python/WMCore/MicroService/Unified/MSTransferor.py

```diff
@@ -41,38 +50,47 @@ def __init__(self, msConfig, logger=None):
         self.uConfig = {}
         self.campaigns = {}
         self.psn2pnnMap = {}
+        # information about RSE/PNNs and their quota/usage
+        self.nodesUsage = {}
+        # list of RSE/PNNs with less than 1TB available, skipped from data placement
+        self.fullRSEs = set()
         self.dsetCounter = 0
         self.blockCounter = 0
 
+    @retry(tries=3, delay=2, jitter=2)
     def updateCaches(self):
```

Comment:
This method is not thread safe, please convert it to make it thread safe. It should get info about caches and return it back to upstream code.